source/mbed_trace.c

```diff
@@ -312,6 +381,16 @@ void mbed_tracef(uint8_t dlevel, const char *grp, const char *fmt, ...)
     mbed_vtracef(dlevel, grp, fmt, ap);
     va_end(ap);
 }
+static uint8_t mbed_trace_get_group_config(const char *grp)
+{
+    // lookup and return found value
+    for (uint32_t pos = 0; pos < m_trace.group_configs_length; pos++) {
+        if (strncmp(grp, m_trace.group_configs[pos].grp, 4) == 0) {
```

Comment:
This adds dependency to new libc function strncmp(), which is a bit unnecessary. Even though the trace group is documented to be max 4 chars, it is not enforced any way in other places, group is just cut when printing it.